src/main/java/io/github/opencubicchunks/cubicchunks/cubicgen/customcubic/CustomGeneratorSettings.java

```diff
@@ -301,237 +375,37 @@ public static CustomGeneratorSettings fromVanilla(ChunkGeneratorSettings setting
-                            {
-                                JsonObject properties = new JsonObject();
-                                properties.add("variant", new JsonPrimitive("stone"));
-                                terrainfill.add("Properties", properties);
-                                terrainfill.add("Name", new JsonPrimitive("minecraft:stone"));
-                            }
-                            JsonObject oceanblock = new JsonObject();
-
-                            {
-                                JsonObject properties = new JsonObject();
-                                properties.add("level", new JsonPrimitive("0"));
-                                oceanblock.add("Properties", properties);
-                                oceanblock.add("Name", new JsonPrimitive("minecraft:water"));
-                            }
-                            defaults.add(MODID + ":biome_fill_depth_offset", new JsonPrimitive(3.0f));
-                            defaults.add(MODID + ":biome_fill_noise_octaves", new JsonPrimitive(4.0f));
-                            defaults.add(MODID + ":height_scale", new JsonPrimitive(factor));
-                            defaults.add(MODID + ":biome_fill_depth_factor", new JsonPrimitive(2.3333333333333335f));
-                            defaults.add(MODID + ":mesa_depth", new JsonPrimitive(16.0f));
-                            defaults.add(MODID + ":water_level", root.get("waterLevel"));
-                            defaults.add(MODID + ":biome_fill_noise_freq", new JsonPrimitive(0.0078125f));
-                        }
-                        replacerConf.add("defaults", defaults);
-                        replacerConf.add("overrides", new JsonObject());
-                    }
-
-                    root.add("replacerConfig", replacerConf);
-                }
-                compound.setString("generatorOptions", gson.toJson(root));
-                return compound;
-            }
-        });
+    public static String fixGeneratorOptions(String generatorOptionsToFix) {
+        generatorOptionsToFix = generatorOptionsToFix.replaceAll("cubicchunks:", MODID + ":");
+        String newGeneratorOptions = new CustomGeneratorSettings().toJson(true);
+        Gson gson = gson(false);
+        
+        JsonReader readerToFix = new JsonReader(new StringReader(generatorOptionsToFix));
+        JsonObject rootToFix = new JsonParser().parse(readerToFix).getAsJsonObject();
 
+        JsonReader newReader = new JsonReader(new StringReader(newGeneratorOptions));
+        JsonObject newRoot = new JsonParser().parse(newReader).getAsJsonObject();
 
-        fixes.registerFix(FixTypes.LEVEL, new IFixableData() {
-            @Override public int getFixVersion() {
-                return 2;
+        for(Entry<String, JsonElement> entry:rootToFix.entrySet()) {
+            if(newRoot.has(entry.getKey())) {
+                newRoot.add(entry.getKey(), entry.getValue());
```

Comment:
This just copies the parts that exist, and ignores the rest. Don't ignore the fact that it should still actually convert old jsons to new ones. I know it takes time to go over old code and figure out what the format used to look like, but it needs to be done, one way or another. 

The way I suggested it to be done was just a suggestion, if you have a better idea, go for it. My idea was something like this (pseudocode, not guaranteed to compile):

```java
newJson.add("v", 3);
newJson.add("waterLevel", getWaterLevel(oldJson));
newJson.add("caves", getCaves(oldJson));
newJson.add("strongholds", getStrongholds(oldJson));
// ...
// ...
newJson.add("standardOres", getStandardOres(oldJson));
// ...
// ...

JsonObject getStandardOres(JsonObject oldJson) {
    JsonObject standardOres = oldJson.get("standardOres");
    if (standardOres == null) {
        standardOres = new JsonSomething();
    }
    if (oldJson.hasKey("ironOreProbability")) {
        // extract old format iron ore
    }
    // ...
    // ...
    return standardOres;
}
```

Yes, it's very verbose. It effectively IS code duplication. It **does** violate the "Don't Repeat Yourself" principle. But this is one of the rare, unusual cases where the code is **NOT** supposed to change together. It **should** be duplicated. One way or another.

Reply:
Can I just reuse your old code instead? You already have a function called `convertStandardOre`.

Reply:
Yes, reusing small parts of it is ok. Just don't copy the whole mess.